docs/dusty-specs/app-specs.md

```diff
@@ -0,0 +1,132 @@
+1. It is specified in different activated app's depends dict. <br/>
+2. It is specified in an activated bundles apps list. <br/>
+As in the depends dict, circular dependencies are still not allowed.
+
+### host_forwarding
+```
+host_forwarding:
+  - host_name: local.website.com
+    host_port: 80
+    container_port: 80
+```
+The host_forwarding key is used specify a list of dictionaries.  Each dictionary allows you to specify a mapping from the containers ip space to your mac's ip space.<br />
+**host_name** is the local url you would like to hit.<br />
+**host_port** is the port on which you will hit the host_name.<br />
+**container_port** is the port the app is running on in the docker container.<br />
+
+### image and build
+```
+build .
+```
+OR
+```
+image: docker.gamechanger.io/gcweb
+```
+The image and build keys are linked because one, but not both, must be specified. These keys are used to specify how to get a base image for this app's container.<br />
+**build** is a path to a Dockerfile's parent directory.<br />
+**image** is a local or remote image.  Dusty (using docker-compose) will attempt to pull the image if it is not found locally.<br />
+Both keys are direct mappings to keys in docker-compose.<br />
+
+### commands
+```
+commands:
+  once:
+    - apt-get update
+    - apt-get install -y nodejs npm
+    - cp /usr/bin/nodejs /usr/bin/node
+    - cd /gc/gcweb/grunt &
+  always:
+    - mkdir -p /var/log/gc
+    - cd /gc/gcweb
+    - python manage.py runserver --noreload
+```
+The commands key is used to specify a list of once and always commands.  These commands are run at different times in the container's life cycle.<br />
+The once commands will only be run the first time an app is started. This mainly occurs if you are running `dusty up`.<br />
+The always commands are run every time the app restarts. This includes on `dusty up`, but also when you do `dusty restart <app_name>`.<br />
+Two sets of commands allows you to do the heavy work of installation in the once command. This frees your restart commands to be very fast and responsive.<br />
+Both the once and always commands are run from an sh file.  This gives you all the flexibility of writing an sh file, including running things in the background (&).<br />
+We are also keeping track of the once and always commands progress using logs automatically created for you in your containers. The always' commands' log is `/var/log/dusty_always.log`. The once's commands' log is `/var/log/dusty_once.log`.<br />
+
+### scripts
+```
+scripts:
+    - name: grunt
+      description: Build assets compiled by Grunt (CSS, JS, that sort of thing)
+      command:
+        - cd /gc/gcweb/grunt
+        - /gc/gcweb/grunt/node_modules/.bin/grunt build
+    - name: grunt-clean
+      description: Remove all compiled files generated by Grunt
+      command:
+        - cd /gc/gcweb/grunt
+        - /gc/gcweb/grunt/node_modules/.bin/grunt clean
```

Comment:
One line command, only need one example. As long as you show the type is a list, they can infer it's okay to have multiple.